backend/baseclass.pm

```diff
@@ -923,14 +923,7 @@ sub similiarity_to_reference ($self, $args) {
     return {sim => $self->reference_screenshot->similarity($self->last_image)};
 }
 
-sub set_tags_to_assert ($self, $args) {
-    my $mustmatch = $args->{mustmatch};
-    my $timeout = $args->{timeout} // $bmwqemu::default_timeout;
-
-    # keep only the most recently used images (https://progress.opensuse.org/issues/15438)
-    needle::clean_image_cache();
-
-    # get the array reference to all matching needles
+sub matching_needles_and_tags ($mustmatch) {
```

Comment:
Personally I use adjectives for properties of objects and verbs for functions like this here. How about calling this function `find_needles_with_tags`?